extensions/amp-timeago/0.2/timeago.js

```diff
@@ -0,0 +1,63 @@
+/**
+ * Copyright 2019 The AMP HTML Authors. All Rights Reserved.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS-IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+import {createElement, useRef} from '../../../src/preact';
+import {timeago} from '../../../third_party/timeagojs/timeago';
+import {useIsIntersecting} from '../../../src/preact/use-is-intersecting';
+import {useResourcesNotify} from '../../../src/preact/utils';
+
+/**
+ * @param {!JsonObject} props
+ * @return {Preact.Renderable}
+ */
+export function Timeago(props) {
+  const timestampRef = useRef(getFuzzyTimestampValue(props));
```

Comment:
We can initialize this with `null`, since it we'll just update it later if we're in viewport.

Reply:
I just confirmed that IntersectionObserver will fire right after calling `observe(node)` if the node is in view.

Reply:
The InOb will fire asynchronously after `observe(node)`, regardless whether the element is in view or not.